internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -150,11 +217,60 @@ func (c *controller) extractValueAsJWTAuthenticator(value authncache.Value) *jwt
 	return jwtAuthenticator
 }
 
-// newJWTAuthenticator creates a jwt authenticator from the provided spec.
-func newJWTAuthenticator(spec *auth1alpha1.JWTAuthenticatorSpec) (*jwtAuthenticator, error) {
-	rootCAs, _, err := pinnipedauthenticator.CABundle(spec.TLS)
+func (c *jwtCacheFillerController) updateStatus(
+	ctx context.Context,
+	original *auth1alpha1.JWTAuthenticator,
+	conditions []*metav1.Condition,
+) error {
+	updated := original.DeepCopy()
+
+	if hadErrorCondition(conditions) {
+		updated.Status.Phase = auth1alpha1.JWTAuthenticatorPhaseError
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonNotReady,
+			Message: "the JWTAuthenticator is not ready: see other conditions for details",
+		})
+	} else {
+		updated.Status.Phase = auth1alpha1.JWTAuthenticatorPhaseReady
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeReady,
+			Status:  metav1.ConditionTrue,
+			Reason:  reasonSuccess,
+			Message: "the JWTAuthenticator is ready",
+		})
+	}
+
+	_ = conditionsutil.MergeConfigConditions(
+		conditions,
+		original.Generation,
+		&updated.Status.Conditions,
+		plog.New().WithName(controllerName),
+		metav1.NewTime(c.clock.Now()),
+	)
+
+	if equality.Semantic.DeepEqual(original, updated) {
+		return nil
+	}
+
+	_, err := c.client.AuthenticationV1alpha1().JWTAuthenticators().UpdateStatus(ctx, updated, metav1.UpdateOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("invalid TLS configuration: %w", err)
+		c.log.Info(fmt.Sprintf("ERROR: %v", err))
```

Comment:
Use the log function's key value pair parameters instead of sprintf for log messages. That way, the log messages will be always have the same main message (with different key/value pairs) and will be machine parsable.

Should this call `c.log.Error` since it is an unexpected error that the admin may need to know about in the logs?

Reply:
Also, this should be covered by a unit test.